### ISSUE TITLE
VSCode hang when Reloading a workspace on Remote SSH connection

### ISSUE THREAD

**Issue by htruongVCC** [microsoft/vscode#225487]
Type: <b>Bug</b>

1) SSH to a remote computer
2) Add Folder to an empty Workspace (that Folder contains a Git repo)
3) VSCode asked for Reload. I clicked on "Reload Window".
4) Then another pop up message saying
" Quitting application is taking a bit longer...
[Quit Anyway]
"
5) Then I clicked on [Quit Anyway], and it hang, maintaing the status "Disconnected from SSH". And from that point of time, I cannot Exit the app. 
Still I can open a new window and work on that new Window. 
And I need to use command KILL in Ubuntu terminal to close the hang VSCode window.
This issue only happened since the latest update. 
Before that, I also frequently clicked on "Reload Window" (as in step 3) when adding Folder to Workspace on an ssh connection, but VSCode didn't hang.

VS Code version: Code 1.92.1 (eaa41d57266683296de7d118f574d0c2652e1fc4, 2024-08-07T20:16:39.455Z)
OS version: Linux x64 6.5.0-45-generic snap
Modes:
Connection to 'SSH' could not be established  Canceled
Remote OS version: Linux x64 5.4.0-148-generic
Remote OS version: Linux x64 5.4.0-150-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-10310U CPU @ 1.70GHz (8 x 3640)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|1, 1, 0|
|Memory (System)|15.39GB (8.86GB free)|
|Process Argv|--no-sandbox --force-user-env --crash-reporter-id dc3327c6-2446-4f9d-97f4-dc3845e1c278|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|wayland|

Connection to 'SSH' could not be established  Canceled

|Item|Value|
|---|---|
|Remote|SSH: 1382|
|OS|Linux x64 5.4.0-148-generic|
|CPUs|Intel(R) Xeon(R) Bronze 3206R CPU @ 1.90GHz (16 x 0)|
|Memory (System)|62.41GB (58.22GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: sga-6|
|OS|Linux x64 5.4.0-150-generic|
|CPUs|Intel(R) Xeon(R) Bronze 3206R CPU @ 1.90GHz (16 x 0)|
|Memory (System)|62.41GB (58.89GB free)|
|VM|0%|
</details>Extensions: none<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
9c06g630:31013171
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-t:31111718

```

</details>

**Comment by htruongVCC** [comment]
![Screenshot from 2024-08-13 14-23-57](https://github.com/user-attachments/assets/e2225c96-1fab-4226-9c63-75def61f87d8)

**Comment by lichq3553** [comment]
the issue fixed after I downgraded to 1.85.2

**Comment by htruongVCC** [comment]
> the issue fixed after I downgraded to 1.85.2

Right. I didn't encounter such problem before.

**Comment by joshspicer** [comment]
Combining as these sounds like similar issues
